ros1_sa_build.sh

```diff
@@ -11,46 +11,5 @@ apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BAD
 apt-get update && apt-get install --no-install-recommends -y python-rosdep python-rosinstall python3-colcon-common-extensions ros-$ROS_DISTRO-ros-base
 pip3 install colcon-bundle colcon-ros-bundle
 
-# Remove the old rosdep sources.list
-rm -rf /etc/ros/rosdep/sources.list.d/*
-rosdep init && rosdep update
-
-. /opt/ros/$ROS_DISTRO/setup.sh
-
-BUILD_DIR_NAME=`basename $TRAVIS_BUILD_DIR`
-
-if [ -z "$WORKSPACES" ]; then
-  WORKSPACES="robot_ws simulation_ws"
-fi
-
-# Run ROSWS update in each workspace before creating archive
-for WS in $WORKSPACES
-do
-  WS_DIR="/${ROS_DISTRO}_ws/src/${BUILD_DIR_NAME}/${WS}"
-  echo "looking for ${WS}, $WS_DIR"
-  if [ -d "${WS_DIR}" ]; then
-    echo "WS ${WS_DIR} found, running rosws update"
-    rosws update -t "${WS_DIR}"
-  fi
-done
-
-# Create archive of all sources files
-SOURCES_INCLUDES="${WORKSPACES} LICENSE* NOTICE* README* roboMakerSettings.json"
-cd /${ROS_DISTRO}_ws/src/${BUILD_DIR_NAME}/
-/usr/bin/zip -r /shared/sources.zip $SOURCES_INCLUDES
-tar cvzf /shared/sources.tar.gz $SOURCES_INCLUDES
-
-for WS in $WORKSPACES
-do
-  # use colcon as build tool to build the workspace if it exists
-  WS_DIR="/${ROS_DISTRO}_ws/src/${BUILD_DIR_NAME}/${WS}"
-  WS_BUILD_SCRIPT="/shared/$(basename ${SCRIPT_DIR})/ws_builds/${WS}.sh"
-  if [ -f "${WS_BUILD_SCRIPT}" ]; then
-    cd "${WS_DIR}"
-    "${WS_BUILD_SCRIPT}"
-    mv ./bundle/output.tar /shared/"${WS}".tar
-  else
-    echo "Unable to find build script ${WS_BUILD_SCRIPT}, build failed"
-    exit 1
-  fi
-done
+COMMON_SA_BUILD_SCRIPT=${SCRIPT_DIR}/common_sa_build.sh
+"${COMMON_SA_BUILD_SCRIPT}"
```

Comment:
To be exactly the same as before, this script should be sourced (e.g. "`. ${COMMON_SA_BUILD_SCRIPT}`") as opposed to executed.

Reply:
Good call, modified.